### PR TITLE
Improve CI

### PR DESCRIPTION
**Pull request by luanpotter**
[trying out new github actions]

### CODE REVIEW

**Comment by luanpotter** [comment]
cc @wolfenrain @fdietze

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -26,82 +30,74 @@ jobs:
...
+         ls "$ANDROID_HOME/tools/bin/"
+         $ANDROID_HOME/tools/bin/sdkmanager "platform-tools" "platforms;android-28"
+         echo "$ANDROID_HOME/platform-tools"
+         ls "$ANDROID_HOME/"
+         ls "$ANDROID_HOME/platform-tools"
+         nohup $ANDROID_HOME/emulator/emulator -avd emu -no-audio -no-snapshot -no-window &
+         $ANDROID_HOME/platform-tools/adb wait-for-device shell 'while [[ -z $(getprop sys.boot_completed | tr -d '\r') ]]; do sleep 1; done; input keyevent 82'
+         $ANDROID_HOME/platform-tools/adb devices
+         echo "Emulator started"
+    #  - name: "Run Flutter Driver tests"
```
Comment (luanpotter) [inline .github/workflows/build.yaml:77]: I will upgrade these tests ASAP so they work with flutter 2

File: .github/workflows/build.yaml [hunk 2]
```diff
@@ -3,6 +3,10 @@ on:
   push:
     branches:
       - master
+  pull_request:
+    branches:
+      - master
+    types: [opened, reopened, synchronize]
```
Comment (erickzanardo) [inline .github/workflows/build.yaml:9]: Just curious why is this needed, if we just omit this types attribute, the action will already run on of those steps right?

**Review by erickzanardo — APPROVED** [review]

File: .github/workflows/build.yaml (hunk 2 above)
Comment (luanpotter) [reply .github/workflows/build.yaml:9]: not sure, this is just how it was before...

**Comment by fdietze** [comment]
I'm happy I kicked this off :+1:

File: .github/workflows/build.yaml [hunk 3]
```diff
@@ -3,6 +3,10 @@ on:
...
 
 jobs:
   build:
```
Comment (wolfenrain) [inline .github/workflows/build.yaml:12]: Should this not be renamed to `web` as it builds the web variant?

File: .github/workflows/build.yaml [hunk 4]
```diff
@@ -26,82 +30,74 @@ jobs:
       - name: Run tests
         run: flutter test
```
Comment (wolfenrain) [inline .github/workflows/build.yaml:31]: I would really put this in it's own seperate stage. Because if your tests are failing there is no need to build your code.

File: .github/workflows/build.yaml (hunk 2 above)
Comment (renancaraujo) [reply .github/workflows/build.yaml:9]: Yep, These three are the default values, we can omit these `types`.

File: .github/workflows/build.yaml (hunk 3 above)
Comment (luanpotter) [reply .github/workflows/build.yaml:12]: well this is actually more than web, it also has the linter, tests, etc

Comment (luanpotter) [reply .github/workflows/build.yaml:11]: I split into two!

File: .github/workflows/build.yaml (hunk 1 above)
Comment (luanpotter) [reply .github/workflows/build.yaml:87]: turns out I can't, @renancaraujo will need to save us

Comment (luanpotter) [reply .github/workflows/build.yaml:87]: that can be a followup

**Review by erickzanardo — APPROVED** [review]
